### PR TITLE
[ONNX] Support onnx if/loop sequence output in opset 13

### PR DESCRIPTION
**Pull request by BowenBao**

### CODE REVIEW

File: torch/csrc/jit/passes/onnx/shape_type_inference.cpp [hunk 1]
```diff
@@ -198,78 +198,89 @@ bool IsSupportedNode(const Node* n) {
...
-      if (scalar_type) {
-        auto v_type = TensorType::create(
-            scalar_type.value(),
-            at::kCPU,
-            c10::SymbolicShape(),
-            c10::VaryingShape<c10::Stride>{},
-            {});
-        input->setType(v_type);
+      case ::c10::prim::ListConstruct: {
+        return CloneValueFromListConstruct(v, n_graph);
```
Comment (jiafatom) [inline torch/csrc/jit/passes/onnx/shape_type_inference.cpp:269]: Is this part a code refactoring? - move it into a function?

File: torch/csrc/jit/passes/onnx/shape_type_inference.cpp [hunk 2]
```diff
@@ -377,6 +388,26 @@ void UpdateOutputTypeByONNXProto(
   }
 }
 
+void FetchBlockInputMetadataFromParent(Block* b) {
+  auto n = b->owningNode();
+  if (nullptr != n && n->kind() == ::c10::onnx::Loop) {
+    // Copy node input metadata to subgraph input.
+    for (size_t i = 0; i < n->inputs().size(); ++i) {
```
Comment (jiafatom) [inline torch/csrc/jit/passes/onnx/shape_type_inference.cpp:395]: auto?

Comment (BowenBao) [reply torch/csrc/jit/passes/onnx/shape_type_inference.cpp:395]: i remember auto triggers clang tidy error, that why switched to size_t

File: torch/csrc/jit/passes/onnx/shape_type_inference.cpp (hunk 1 above)
Comment (BowenBao) [reply torch/csrc/jit/passes/onnx/shape_type_inference.cpp:269]: yes

**Review by jiafatom — APPROVED** [review]

**Comment by BowenBao** [comment]
closing due to move to merge with dev branch https://github.com/pytorch/pytorch/pull/49270

**Pull request closed without merging**